client/my-sites/backup/backup-date-picker/test/date-button.jsx

```diff
@@ -14,35 +17,21 @@ jest.mock( '../hooks', () => ( {
 	useCanGoToDate: jest.fn(),
 } ) );
 
-import { shallow } from 'enzyme';
+import { fireEvent, render, screen } from '@testing-library/react';
 import moment from 'moment';
-import { useDispatch, useSelector } from 'react-redux';
+import { recordTracksEvent } from 'calypso/state/analytics/actions/record';
 import DateButton from '../date-button';
 
-const getTracksEventName = ( event ) => event.meta.analytics[ 0 ].payload.name;
-
 describe( 'Test Backup Date Picker', () => {
 	beforeEach( () => {
 		jest.clearAllMocks();
-
-		useDispatch.mockImplementation( mockUseDispatch );
-		useSelector.mockImplementation( mockUseSelector );
 	} );
+	test( 'Records a tracks event when the date picker is opened', () => {
+		render( <DateButton selectedDate={ moment() }></DateButton> );
 
-	test( 'Records a tracks event when the date picker is opened', () =>
-		new Promise( ( done ) => {
-			const checkTracksEvent = ( event ) => {
-				const name = getTracksEventName( event );
-				expect( name ).toEqual( 'calypso_jetpack_backup_date_picker_open' );
-				done();
-			};
-
-			// mock the useDispatch hook that will pick up the tracks event
-			useDispatch.mockImplementation( () => checkTracksEvent );
-
-			// shallow render
-			const DatePicker = shallow( <DateButton selectedDate={ moment() } /> );
-			// simulate a click on the button
-			DatePicker.find( '.backup-date-picker__date-button-button' ).simulate( 'click' );
-		} ) );
+		fireEvent.click( screen.getByText( 'Select Date' ) );
```

Comment:
Using [`@testing-library/user-event`](https://testing-library.com/docs/react-testing-library/migrate-from-enzyme/#simulate-user-events) would be preferred over `fireEvent`. In this case you'd use it like

```js
import userEvent from '@testing-library/user-event';

// ... then inside your test function
await userEvent.click( screen.getByText( 'Select Date' );

// ... or using the `setup` explicitly
const user = userEvent.setup();
await user.click( screen.getByText( 'Select Date' );
```
To me the latter makes more sense if you have more than one interaction in your test fn. Using `user-event`  does a bunch more than firing only a single event and is closer to how an actual user would interact with a component.

Reply:
Thanks for the suggestion @flootr , I have updated to use `userEvent`